src/repositories/user.repository.ts

```diff
@@ -30,4 +41,41 @@ export class UserRepository extends DefaultCrudRepository<
       async () => orderRepository,
     );
   }
+
+  async login(credentials: Credentials) {
+    // Validate Email
+    if (!isemail.validate(credentials.email)) {
+      throw new HttpErrors.UnprocessableEntity('invalid email');
+    }
+
+    // Validate Password Length
+    if (credentials.password.length < 8) {
+      throw new HttpErrors.UnprocessableEntity(
+        'password must be minimum 8 characters',
+      );
+    }
+
+    // Check if user exists
+    try {
+      const foundUser = await this.findOne({
+        where: {email: credentials.email, password: credentials.password},
+      });
+
+      let token = undefined;
+      if (foundUser) {
+        const currentUser = _.pick(toJSON(foundUser), ['id', 'email']);
+        // Generate user token using JWT
+        token = await signAsync(currentUser, 'secretforjwt', {
+          expiresIn: 300,
+        });
+      } else {
+        return Promise.reject(
```

Comment:
Let's rewrite this block as follows - the code below is just an illustration, may not work OOTB.

```ts
const foundUser = await this.findOne({
  where: {email: credentials.email, password: credentials.password},
});

if (!foundUser) {
  new HttpErrors.Unauthorized('Wrong credentials!')
}

const currentUser = _.pick(toJSON(foundUser), ['id', 'email']);

// Generate user token using JWT
token = await signAsync(currentUser, 'secretforjwt', {
  expiresIn: 300,
});

return token;
```

See the next two comments for explanation.